File: tests/benchmarks/ws/echo_server.py
#!/usr/bin/env python3
"""
echo_server.py - Python WebSocket echo server (websockets) for benchmark.

Usage:
  python3 echo_server.py [port]

Default port: 9001
Requires: pip3 install websockets
"""

import asyncio
import sys

import websockets


async def echo(websocket):
    async for message in websocket:
        await websocket.send(message)


async def main():
    port = int(sys.argv[1]) if len(sys.argv) > 1 else 9001
    print(f"Python WebSocket Echo Server (websockets) on port {port}")
    async with websockets.serve(
        echo, "127.0.0.1", port, ping_interval=None, max_size=10 * 1024 * 1024
    ):
        await asyncio.Future()


if __name__ == "__main__":
    asyncio.run(main())

File: tests/benchmarks/ws/echo_server.js
// echo_server.js - Node.js WebSocket echo server (ws library) for benchmark.
//
// Usage:
//   node echo_server.js [port]
//
// Default port: 9001
// Requires: npm install ws

const WebSocket = require("ws");

const port = parseInt(process.argv[2] || "9001", 10);

const wss = new WebSocket.Server({
  port: port,
  host: "127.0.0.1",
  perMessageDeflate: false,
  maxPayload: 10 * 1024 * 1024,
});

wss.on("listening", () => {
  console.log(`Node.js WebSocket Echo Server (ws) on port ${port}`);
});

wss.on("connection", (ws) => {
  ws.on("message", (data, isBinary) => {
    ws.send(data, { binary: isBinary });
  });
});

wss.on("error", (err) => {
  console.error("Server error:", err);
  process.exit(1);
});
